src/otg/dataset/summary_statistics.py

```diff
@@ -122,14 +123,71 @@ def pvalue_filter(self: SummaryStatistics, pvalue: float) -> SummaryStatistics:
         )
         return SummaryStatistics(_df=df, _schema=self._schema)
 
-    def window_based_clumping(self: SummaryStatistics, distance: int) -> StudyLocus:
-        """Perform distance-based clumping.
+    def window_based_clumping_with_locus(
+        self: SummaryStatistics,
+        distance: int,
+        gwas_signiciance: float = 5e-8,
+        baseline_significance: float = 0.05,
+    ) -> StudyLocus:
+        """Generate study-locus from summary statistics by distance based clumping + collect locus.
 
         Args:
-            distance (int): Distance in base pairs
+            self (SummaryStatistics): _description_
+            distance (int): _description_
+            gwas_signiciance (float, optional): _description_. Defaults to 5e-8.
+            baseline_significance (float, optional): _description_. Defaults to 0.05.
 
         Returns:
-            StudyLocus: StudyLocus object
+            StudyLocus: _description_
         """
         # Calculate distance-based clumping:
-        return WindowBasedClumping.clump(self, distance)
+        return WindowBasedClumping.clump_with_locus(
+            self,
+            window_length=distance,
+            p_value_significance=gwas_signiciance,
+            p_value_baseline=baseline_significance,
+        )
+
+    def window_based_clumping(
+        self: SummaryStatistics, distance: int, gwas_signiciance: float = 5e-8
+    ) -> StudyLocus:
+        """Generate study-locus from summary statistics by distance based clumping + collect locus.
+
+        Args:
+            self (SummaryStatistics): _description_
+            distance (int): _description_
+            gwas_signiciance (float, optional): _description_. Defaults to 5e-8.
+
+        Returns:
+            StudyLocus: _description_
+        """
+        # Calculate distance-based clumping:
+        return WindowBasedClumping.clump(
+            self, window_length=distance, p_value_significance=gwas_signiciance
+        )
+
+    def exclude_region(self: SummaryStatistics, region: str) -> SummaryStatistics:
```

Comment:
I was also thinking about that, however to be precise with that, we should have something like a region object as hail has. I'm not 100% sure about that.

Reply:
There are different ways of improving this. One option to comes to my mind, it's similar to how we handle the QC's in GWASCatalog. A class that contains a list of predefined regions we concluded are potentially worth excluding. It won't be flexible, but it would ensure things are readable.

In any case, this is not critical for this PR